core/src/main/java/org/apache/calcite/avatica/util/DateTimeUtils.java

```diff
@@ -993,7 +993,7 @@ public static int addMonths(int date, int m) {
     return DateTimeUtils.ymdToUnixDate(y0, m0, d0);
   }
 
-  private static int lastDay(int y, int m) {
+  public static int lastDay(int y, int m) {
```

Comment:
Making the method public requires some javadoc. Can you please add a short description? 
Although not strictly necessary adding tests will be a plus.